rotkehlchen/tests/utils/blockchain.py

```diff
@@ -611,7 +616,18 @@ def setup_evm_addresses_activity_mock(
         polygon_pos_addresses: Optional[list[ChecksumEvmAddress]] = None,
         arbitrum_one_addresses: Optional[list[ChecksumEvmAddress]] = None,
         base_addresses: Optional[list[ChecksumEvmAddress]] = None,
+        gnosis_addresses: Optional[list[ChecksumEvmAddress]] = None,
 ) -> 'ExitStack':
+    mocked_addresses = {
```

Comment:
Not terrible since it works, but the locals() aproach I suggested is better in the sense you don't repeat the same stuff in 8 lines and need to change it for every new argument here.

Instead just save the locals so they are visible in the mock function.

```
-        ethereum_addresses: list[ChecksumEvmAddress],
+        ethereum_addresses: list[ChecksumEvmAddress],  # pylint: disable=unused-argument  # used by the saved locals  # noqa: E501, RUF100
         avalanche_addresses: Optional[list[ChecksumEvmAddress]] = None,
-        optimism_addresses: Optional[list[ChecksumEvmAddress]] = None,
-        polygon_pos_addresses: Optional[list[ChecksumEvmAddress]] = None,
-        arbitrum_one_addresses: Optional[list[ChecksumEvmAddress]] = None,
-        base_addresses: Optional[list[ChecksumEvmAddress]] = None,
-        gnosis_addresses: Optional[list[ChecksumEvmAddress]] = None,
+        optimism_addresses: Optional[list[ChecksumEvmAddress]] = None,  # pylint: disable=unused-argument  # used by the saved locals  # noqa: E501, RUF100
+        polygon_pos_addresses: Optional[list[ChecksumEvmAddress]] = None,  # pylint: disable=unused-argument  # used by the saved locals  # noqa: E501, RUF100
+        arbitrum_one_addresses: Optional[list[ChecksumEvmAddress]] = None,  # pylint: disable=unused-argument  # used by the saved locals  # noqa: E501, RUF100
+        base_addresses: Optional[list[ChecksumEvmAddress]] = None,  # pylint: disable=unused-argument  # used by the saved locals  # noqa: E501, RUF100
+        gnosis_addresses: Optional[list[ChecksumEvmAddress]] = None,  # pylint: disable=unused-argument  # used by the saved locals  # noqa: E501, RUF100
 ) -> 'ExitStack':
-    mocked_addresses = {
-        'ethereum_addresses': ethereum_addresses,
-        'avalanche_addresses': avalanche_addresses,
-        'optimism_addresses': optimism_addresses,
-        'polygon_pos_addresses': polygon_pos_addresses,
-        'arbitrum_one_addresses': arbitrum_one_addresses,
-        'base_addresses': base_addresses,
-        'gnosis_addresses': gnosis_addresses,
-    }
+    saved_locals = locals()  # bit hacky, but save locals here so they can be accessed by mock_chain_has_activity  # noqa: E501
```